File: leetcode/830.positions-of-large-groups.cpp
class Solution {
public:
    vector<vector<int>> largeGroupPositions(string S) {
        vector<vector<int>> ret;
        int count = 1;
        int start = 0;
        int len = S.length();
        for (int i = 1; i < len; ++i) {

            if (S[i] == S[i - 1]) {
                ++count;
                continue;
            }
            if (count < 3) {
                count = 1;
                start = i;
                continue;
            }
            vector<int> v;
            v.push_back(start);
            v.push_back(i - 1);
            ret.push_back(v);

            start = i;
            count = 1;
        }
        if (count >= 3) {
            vector<int> v;
            v.push_back(start);
            v.push_back(len - 1);
            ret.push_back(v);
        }
        return ret;
    }
};

File: leetcode/830.positions-of-large-groups.go
package leetcode
func largeGroupPositions(s string) [][]int {
  var ans [][]int
  var cur rune
  start := -1
  for pos, char := range s {
    if char != cur {
      if pos - start >= 3 {
        ans = append(ans, []int{start, pos - 1})
      }
      cur = char
      start = pos
    }
  }
  if len(s) - start >= 3 {
    ans = append(ans, []int{start, len(s) - 1})
  }
  return ans
}